be/src/column/binary_column.cpp

```diff
@@ -209,6 +209,23 @@ void BinaryColumnBase<T>::_build_slices() const {
     _slices_cache = true;
 }
 
+template <typename T>
+void BinaryColumnBase<T>::fill_default(const Filter& filter) {
+    std::vector<uint32_t> indexes;
+    for (size_t i = 0; i < filter.size(); i++) {
+        Slice slice = get_slice(i);
+        if (filter[i] == 1 && slice.size > 0) {
+            indexes.push_back(i);
+        }
+    }
+    if (indexes.empty()) {
+        return;
+    }
+    auto default_column = clone_empty();
+    default_column->append_default(indexes.size());
```

Comment:
can we move not-null items to the front of byte array compactly?

Reply:
How?